### PR TITLE
Provider function to replace validation.NoZeroValues 

### LINKED ISSUE #2462: Crash azurerm_route_table name with only spaces in it
The `azurerm_route_table` data source is crashing `terraform` when a name consisting only of spaces is provided.

Yes, if it hurts don't do it.  ;-) . Thing is that this can make a simple bug difficult to figure out, as it was in my case. (Had to do with a string that was being processed to calculate the name. In this instance there was an error in processing resulting in a space in the name.)

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v0.11.10
azurerm" (1.19.0)
```

### Affected Resource(s)

* Data source `azurerm_route_table`

### Terraform Configuration Files

```hcl
provider "azurerm" {
}

resource "azurerm_resource_group" "test" {
  name = "testResourceGroup"
  location = "West US"
}

data "azurerm_route_table" "source" {
  resource_group_name = "${azurerm_resource_group.test.name}"
  # Note the space ( ) in the name.
  name = " "
}
```

### Debug Output

[Output Gist](https://gist.github.com/pedrohdz/0e5287b2fbdd8634a7af9475a8cf1d3c)

### Panic Output

[crash.log](https://gist.github.com/pedrohdz/2f1b45944625a969844d6cf2068213e0)

### Expected Behavior

Anything but a panic?  A validation error would be the way to go.

### Actual Behavior

```
panic: runtime error: invalid memory address or nil pointer dereference
```

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

### Important Factoids

**Nope!**

### References

It seems like the issue is cropping up to the use of `validation.NoZeroValues` in [data_source_route_table.go](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/data_source_route_table.go#L20)?  Might be better to use one of the regex validators?

### PR DESCRIPTION
**Pull request by jeffreyCline**
Fixes #2462

### CODE REVIEW

File: azurerm/data_source_route_table.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ func dataSourceArmRouteTable() *schema.Resource {
 			"name": {
 				Type:         schema.TypeString,
 				Required:     true,
-				ValidateFunc: validation.NoZeroValues,
+				ValidateFunc: validate.NoEmptyStrings(),
```
Comment (katbyte) [inline azurerm/data_source_route_table.go]: We shouldn't need to call the function:
```suggestion
				ValidateFunc: validate.NoEmptyStrings,
```

File: azurerm/helpers/validate/strings.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+func NoEmptyStrings() schema.SchemaValidateFunc {
+	return func(i interface{}, k string) ([]string, []error) {
+		v, ok := i.(string)
+		if !ok {
+			return nil, []error{fmt.Errorf("expected type of %q to be string", k)}
+		}
+
+		newv := strings.TrimSpace(v)
+
+		if len(newv) == 0 {
```
Comment (katbyte) [inline azurerm/helpers/validate/strings.go]: this could be:

```go
if strings.TrimSpace(v) == "" {
```

File: azurerm/resource_arm_route_table.go [hunk 3]
```diff
@@ -30,7 +31,7 @@ func resourceArmRouteTable() *schema.Resource {
 				Type:         schema.TypeString,
 				Required:     true,
 				ForceNew:     true,
-				ValidateFunc: validation.NoZeroValues,
+				ValidateFunc: validate.NoEmptyStrings(),
```
Comment (katbyte) [inline azurerm/resource_arm_route_table.go]:
```suggestion
				ValidateFunc: validate.NoEmptyStrings,
```

File: azurerm/resource_arm_route_table.go [hunk 4]
```diff
@@ -46,13 +47,13 @@ func resourceArmRouteTable() *schema.Resource {
 						"name": {
 							Type:         schema.TypeString,
 							Required:     true,
-							ValidateFunc: validation.NoZeroValues,
+							ValidateFunc: validate.NoEmptyStrings(),
```
Comment (katbyte) [inline azurerm/resource_arm_route_table.go]:
```suggestion
							ValidateFunc: validate.NoEmptyStrings,
```

File: azurerm/resource_arm_route_table.go [hunk 5]
```diff
@@ -46,13 +47,13 @@ func resourceArmRouteTable() *schema.Resource {
...
 						},
 
 						"address_prefix": {
 							Type:         schema.TypeString,
 							Required:     true,
-							ValidateFunc: validation.NoZeroValues,
+							ValidateFunc: validate.NoEmptyStrings(),
```
Comment (katbyte) [inline azurerm/resource_arm_route_table.go]:
```suggestion
							ValidateFunc: validate.NoEmptyStrings,
```

File: azurerm/resource_arm_route_table.go [hunk 6]
```diff
@@ -71,7 +72,7 @@ func resourceArmRouteTable() *schema.Resource {
 						"next_hop_in_ip_address": {
 							Type:         schema.TypeString,
 							Optional:     true,
-							ValidateFunc: validation.NoZeroValues,
+							ValidateFunc: validate.NoEmptyStrings(),
```
Comment (katbyte) [inline azurerm/resource_arm_route_table.go]:
```suggestion
							ValidateFunc: validate.NoEmptyStrings,
```

File: azurerm/helpers/validate/strings_test.go [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+		{
+			Value:    "  1",
+			TestName: "DoubleSpaceOne",
+			ErrCount: 0,
+		},
+		{
+			Value:    "1 ",
+			TestName: "OneSpace",
+			ErrCount: 0,
+		},
```
Comment (tombuildsstuff) [inline azurerm/helpers/validate/strings_test.go]: it'd be good to add a test case for 5 empty spaces in here too

Comment (jeffreyCline) [reply azurerm/helpers/validate/strings_test.go]: Doen.
